pincer/objects/guild/channel.py

```diff
@@ -661,6 +662,347 @@ async def create_invite(
+        -------
+        AsyncIterator[:class:`~.pincer.objects.channel.ThreadMember`]
+            An iterator of thread members.
+        """
+        data = await self._http.get(f"channels/{self.id}/thread-members")
+        for member in data:
+            yield ThreadMember.from_dict(
+                construct_client_dict(self._client, member)
+            )
+
+    async def list_active_threads(self) -> dict[str, Any]:
+        """
+        Returns all active threads in the channel, including public and
+        private threads. Threads are ordered by their id, in descending order.
+
+        Returns
+        -------
+        dict[:class:`str`, :class:`Any`]
+            Dictionary containing the response body:
+                - threads: Generator of the active threads.
+                - members: Generator for each thread member object for each
+                returned thread the current user has joined.
+                - has_more: Whether there are potentially additional threads
+                that could be returned on a subsequent call.
+        """
+        data = self._http.get(f"channels/{self.id}/threads/active")
+        return {
+            "threads": (Channel.from_dict(construct_client_dict(
+                self._client, t
+            )) for t in data["threads"]),
+            "members": (ThreadMember.from_dict(construct_client_dict(
+                self._client, m
+            )) for m in data["members"]),
+            "has_more": data["has_more"]
+        }
+
+    async def list_public_archived_threads(
+        self,
+        before: Timestamp = None,
+        limit: int = None
+    ) -> dict[str, Any]:
+        """
+        Returns archived threads in the channel that are public.
+        When called on a ``GUILD_TEXT`` channel, returns threads of type
+        ``GUILD_PUBLIC_THREAD``. When called on a ``GUILD_NEWS`` channel returns
+        threads of type ``GUILD_NEWS_THREAD``. Threads are ordered by
+        ``archive_timestamp``, in descending order. Requires the
+        ``READ_MESSAGE_HISTORY`` permission.
+
+        Parameters
+        ----------
+        before: Optional[:class:`~.pincer.objects.timestamp.Timestamp`]
+            Returns threads before this timestamp.
+        limit: Optional[:class:`int`]
+            The maximum number of threads to return.
+
+        Returns
+        -------
+        dict[:class:`str`, :class:`Any`]
+            Dictionary containing the response body:
+                - threads: Generator of the public archived threads.
+                - members: Generator for each thread member object for each
+                returned thread the current user has joined.
+                - has_more: Whether there are potentially additional threads
+                that could be returned on a subsequent call.
+        """
+        data = self._http.get(
+            f"channels/{self.id}/threads/archived/public",
+            data={"before": before, "limit": limit}
+        )
```

Comment:
According to the [Discord Developer Portal](https://discord.com/developers/docs/resources/channel#list-public-archived-threads), both of this should be optional though. Also, would not `remove_none()` work here?

Reply:
1. `http.get` does not have a data argument.
2. The data arguments send things in the body of the request, not as a query string. (maybe in the future I'll add a query string parameter)
3. `remove_none` won't work because it's not a dictionary.
4. Just repeat this for each argument after the url `+ (f"&{NAME=}" if NAME else "")`

Reply:
new solution. just committed a function to convert a Dict to a query string. It's in the latest branch and called `dict_to_query`

Reply:
well. removing that because you can use `urlencode` from `urllib.parse` instead. 💀